types/header.go

```diff
@@ -54,6 +54,11 @@ func (n Nonce) String() string {
 	return hex.EncodeToHex(n[:])
 }
 
+// MarshalText implements encoding.TextMarshaler
+func (n Nonce) MarshalText() ([]byte, error) {
+	return []byte(n.String()), nil
+}
+
```

Comment:
Nitpick:

could we remove `Nonce` type outside of Header implementation (below `Header` methods implementation and above `Block`)? It is strange having it interleaved.

Reply:
Yes, I have similar thoughts. Do not want to touch that yet.